From eadac226f6c4475c3619313057fdca75cb79441a Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Wed, 17 Jun 2020 06:33:51 -0700
Subject: [PATCH 1408/1851] soc: xilinx: vcu: Add proper error checking

commit b2499ee9aa216ab59d8669bc59f4ef63abd4f5ee from
https://github.com/Xilinx/linux-xlnx.git

If probe() fails due to unavailable resources, it returns with
-EPROBE_DEFER and retries after sometime, so no need to print error
message.

Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/soc/xilinx/xlnx_vcu.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/soc/xilinx/xlnx_vcu.c b/drivers/soc/xilinx/xlnx_vcu.c
index d89fcbaf9446..b529b1126e0f 100644
--- a/drivers/soc/xilinx/xlnx_vcu.c
+++ b/drivers/soc/xilinx/xlnx_vcu.c
@@ -267,31 +267,36 @@ static int xvcu_probe(struct platform_device *pdev)
 
 	xvcu->pll_ref = devm_clk_get(pdev->dev.parent, "pll_ref");
 	if (IS_ERR(xvcu->pll_ref)) {
-		dev_err(&pdev->dev, "Could not get pll_ref clock\n");
+		if (PTR_ERR(xvcu->pll_ref) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Could not get pll_ref clock\n");
 		return PTR_ERR(xvcu->pll_ref);
 	}
 
 	xvcu->core_enc = devm_clk_get(pdev->dev.parent, "vcu_core_enc");
 	if (IS_ERR(xvcu->core_enc)) {
-		dev_err(&pdev->dev, "Could not get core_enc clock\n");
+		if (PTR_ERR(xvcu->core_enc) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Could not get core_enc clock\n");
 		return PTR_ERR(xvcu->core_enc);
 	}
 
 	xvcu->core_dec = devm_clk_get(pdev->dev.parent, "vcu_core_dec");
 	if (IS_ERR(xvcu->core_dec)) {
-		dev_err(&pdev->dev, "Could not get vcu_core_dec clock\n");
+		if (PTR_ERR(xvcu->core_dec) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Could not get core_dec clock\n");
 		return PTR_ERR(xvcu->core_dec);
 	}
 
 	xvcu->mcu_enc = devm_clk_get(pdev->dev.parent, "vcu_mcu_enc");
 	if (IS_ERR(xvcu->mcu_enc)) {
-		dev_err(&pdev->dev, "Could not get mcu_enc clock\n");
+		if (PTR_ERR(xvcu->mcu_enc) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Could not get mcu_enc clock\n");
 		return PTR_ERR(xvcu->mcu_enc);
 	}
 
 	xvcu->mcu_dec = devm_clk_get(pdev->dev.parent, "vcu_mcu_dec");
 	if (IS_ERR(xvcu->mcu_dec)) {
-		dev_err(&pdev->dev, "Could not get mcu_dec clock\n");
+		if (PTR_ERR(xvcu->mcu_dec) != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "Could not get mcu_dec clock\n");
 		return PTR_ERR(xvcu->mcu_dec);
 	}
 
-- 
2.31.1

